### PR TITLE
issue-2530-system-collections-generic-stack

### PR DESCRIPTION
**Pull request by njlr**
* Adds Stack and various tests

### CODE REVIEW

**Comment by Happypig375** [comment]
Dynamic resizing looks like ResizeArray's job.

**Comment by njlr** [comment]
> Dynamic resizing looks like ResizeArray's job.

Many ways to do this I'm sure. This implementation is based on the one in [FCS](https://github.com/dotnet/fsharp/blob/main/src/fsharp/utils/prim-parsing.fs#L88-L111) and extended to match the `System.Collections.Generic.Stack` API. If there's a reason to leverage `ResizeArray` I can update it.

File: src/Fable.Transforms/Replacements.fs [hunk 1]
```diff
@@ -2474,6 +2474,17 @@ let hashSets (com: ICompiler) (ctx: Context) r t (i: CallInfo) (thisArg: Expr op
...
+let stacks (com: ICompiler) (ctx: Context) r t (i: CallInfo) (thisArg: Expr option) (args: Expr list) =
+    match i.CompiledName, thisArg, args with
+    | ".ctor", None, args ->
+        let moduleName, mangledName = getMangledNames i thisArg
+        Helper.LibCall(com, moduleName, mangledName, t, args, i.SignatureArgTypes, isJsConstructor=true, ?thisArg=thisArg, ?loc=r) |> Some
+    | ("Push" | "Pop" | "Peek" | "get_Count" | "Clear" | "Contains" |
+        "TryPeek" | "TryPop" | "ToArray" as meth), Some c, args ->
+        let moduleName, mangledName = getMangledNames i thisArg
+        Helper.LibCall(com, moduleName, mangledName, t, args, i.SignatureArgTypes, ?thisArg=thisArg, ?loc=r) |> Some
+    | _ -> None
```
Comment (alfonsogarciacaro) [inline src/Fable.Transforms/Replacements.fs:2486]: We have the `bclType` function to automatically map  types from the BCL (see `System.Text.StringBuilder` for example). Maybe could be used here too?

**Review by alfonsogarciacaro — APPROVED** [review]

**Comment by alfonsogarciacaro** [comment]
Yes, `ResizeArray` should work too, but it's a good idea to keep the implementation as close as possible to FCS to reduce maintenance. I've also got good results when preallocating arrays in JS and this may help when compiling to other languages too.

File: src/Fable.Transforms/Replacements.fs (hunk 1 above)
Comment (njlr) [reply src/Fable.Transforms/Replacements.fs]: Yep, `bclType` worked nicely!

**Comment by alfonsogarciacaro** [comment]
Thanks!

**Pull request merged**